crates/interpreter/src/instructions/opcode.rs

```diff
@@ -52,17 +53,17 @@ macro_rules! opcodes {
         }
 
         // in `eval`:
-        (const { make_instruction_table::<SPEC>() })[opcode as usize](interpreter, host)
+        (const { make_instruction_table::<H, SPEC>() })[opcode as usize](interpreter, host)
         */
 
         /// Evaluates the opcode in the given context.
         #[inline(always)]
-        pub(crate) fn eval<SPEC: Spec>(opcode: u8, interpreter: &mut Interpreter, host: &mut dyn Host) {
+        pub(crate) fn eval<H: Host, SPEC: Spec>(opcode: u8, interpreter: &mut Interpreter, host: &mut H) {
```

Comment:
```suggestion
        pub fn eval<H: Host, SPEC: Spec>(opcode: u8, interpreter: &mut Interpreter, host: &mut H) {
```

Not related but would be good to be `public`